backend/src/db-utils.js

```diff
@@ -1111,20 +1111,99 @@ const queryExecutedReceiptsList = async (blockHash) => {
 };
 
 const queryContractInfo = async (accountId) => {
-  return await querySingleRow(
-    [
-      `SELECT
-        DIV(receipts.included_in_block_timestamp, 1000*1000) AS block_timestamp,
-        receipts.originated_from_transaction_hash AS hash
-       FROM action_receipt_actions
-       LEFT JOIN receipts ON receipts.receipt_id = action_receipt_actions.receipt_id
-       WHERE action_receipt_actions.action_kind = 'DEPLOY_CONTRACT' AND action_receipt_actions.receipt_receiver_account_id = :account_id
-       ORDER BY action_receipt_actions.receipt_included_in_block_timestamp DESC
-       LIMIT 1`,
-      { account_id: accountId },
-    ],
-    { dataSource: DS_INDEXER_BACKEND }
-  );
+  let contractInfo;
+  // query to analytics db to find latest historical record
+  const queryLatestHistoricalInfo = async () => {
```

Comment:
1. Fix the order of requests as per offline discussion
2. Avoid unnecessary functions